src/extension-runners/firefox-android.js

```diff
@@ -122,15 +123,29 @@ export class FirefoxAndroidExtensionRunner {
       adbHost,
       adbPort,
       ADBUtils = DefaultADBUtils,
+      cleanArtifacts,
     } = this.params;
 
+
     this.adbUtils = new ADBUtils({
       adbBin, adbHost, adbPort,
     });
+    // Call all the registered cleanup callbacks.
+    for (const fn of this.cleanupCallbacks) {
+      try {
+        fn();
+      } catch (error) {
+        log.error(error);
+      }
+    }
 
     await this.adbDevicesDiscoveryAndSelect();
     await this.apkPackagesDiscoveryAndSelect();
     await this.adbCheckRuntimePermissions();
+    if (cleanArtifacts) {
+      await this.adbClearArtifactsDir();
+    }
+    await this.adbCheckArtifactsDir();
```

Comment:
could you explain me the rationale of this?
e.g.
- who is going to pass this `cleanArtifacts` set to `true`
- what are the roles of `adbClearArtifactsDir` and `adbCheckArtifactsDir` in the approach you are thinking of?

Reply:
This for cli comment if user sends it then first clear artifact otherwise if user doesn't sends cli option then just check them and give info or warning to them as suggested below